ipatests/test_integration/test_sssd.py

```diff
@@ -108,3 +114,57 @@ def test_auth_cache_enabled_when_configured(self, user):
             assert self.is_auth_cached(self.users[user])
             time.sleep(10)
             assert not self.is_auth_cached(self.users[user])
+
+    @contextmanager
+    def filter_user_setup(self, user, provider='ipa'):
+        tasks.kinit_admin(self.master)
+        master = self.master
+        sssd_conf_backup = tasks.FileBackup(master, paths.SSSD_CONF)
+        if user['name'] == 'fakeuser':
+            provider = 'nss'
+        filter = {'filter_users': '{0}'.format(user['name'])}
+        try:
+            tasks.modify_sssd_conf(
+                master, master.domain.name, filter, provider)
+            yield
+        finally:
+            sssd_conf_backup.restore()
+            tasks.clear_sssd_cache(self.master)
+
+    def is_user_filtered(self, user):
+        log_file = '{0}/sssd_nss.log'.format(paths.VAR_LOG_SSSD_DIR)
+        logsize = tasks.get_logsize(self.master, log_file)
+        try:
+            self.master.run_command(['getent', 'passwd', user['name']],
+                                    raiseonerr=False)
+            test_id = self.master.run_command(['id', user['name']],
+                                              raiseonerr=False)
+            assert test_id.returncode != 0
+        finally:
+            sssd_log = self.master.get_file_contents(log_file)[logsize:]
+
+        dp_req = ("Looking up [{0}] in data provider".format(user['name']))
+        return dp_req.encode() in sssd_log
+
+    @pytest.mark.xfail(
+        osinfo.id == 'fedora' and osinfo.version_number <= (30,),
+        reason='https://pagure.io/SSSD/sssd/issue/3978', strict=False)
```

Comment:
Please do not use xfail with strict=False unless you have a serious reason for that.
Doing so you simply ignore test result, so that we shall not notice if it is finally fixed in Fedora 30 and older